From 7be432057f268fff6e68928f3c3f3b4b70d67368 Mon Sep 17 00:00:00 2001
From: Kenny Levinsen <kl@kl.wtf>
Date: Fri, 30 Apr 2021 20:02:02 +0200
Subject: [PATCH] meson: Copy seatd's scdoc handling

This fixes cross-compiling, and avoids unnecessary search for a shell.
---
 meson.build | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/meson.build b/meson.build
index 23d24c5..9a02caf 100644
--- a/meson.build
+++ b/meson.build
@@ -3,7 +3,7 @@ project(
 	'c',
 	version: '0.2.0',
 	license: 'MIT',
-	meson_version: '>=0.53.0',
+	meson_version: '>=0.56.0',
 	default_options: [
 		'c_std=c11',
 		'warning_level=3',
@@ -56,16 +56,13 @@ executable(
 	install: true,
 )
 
-scdoc = dependency('scdoc', required: get_option('man-pages'), version: '>= 1.9.7')
+scdoc = dependency('scdoc', required: get_option('man-pages'), version: '>= 1.9.7', native: true)
 
 if scdoc.found()
-	sh = find_program('sh')
-
-	man_pages = ['wlsunset.1.scd']
-
+	scdoc_prog = find_program(scdoc.get_variable(pkgconfig: 'scdoc'), native: true)
 	mandir = get_option('mandir')
 
-	foreach src : man_pages
+	foreach src : ['wlsunset.1.scd']
 		topic = src.split('.')[0]
 		section = src.split('.')[1]
 		output = '@0@.@1@'.format(topic, section)
@@ -75,7 +72,7 @@ if scdoc.found()
 			input: src,
 			output: output,
 			command: [
-				sh, '-c', '@0@ < @INPUT@ > @1@'.format(scdoc.get_pkgconfig_variable('scdoc'), output)
+				'sh', '-c', '@0@ < @INPUT@ > @1@'.format(scdoc_prog.full_path(), output)
 			],
 			install: true,
 			install_dir: '@0@/man@1@'.format(mandir, section)
-- 
2.30.2

